core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/CloneWorkspaceContainerRunner.scala

```diff
@@ -0,0 +1,127 @@
+      t match {
+        case Some(t) => logger.error(logMessage, t)
+        case None    => logger.error(logMessage)
+      }
+    }
+
+    val workspaceId = job.workspaceId match {
+      case Some(name) => name
+      case None =>
+        logFailure("no workspace id set")
+        return Future.successful(Complete) // nothing more this runner can do with it
+    }
+
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        val msg =
+          s"Unable to update clone status for workspace $workspaceId because no user email set on monitoring job"
+        logFailure(msg)
+        return cloneFail(workspaceId, msg).map(_ => Complete)
+    }
+
+    getUserCtx(userEmail).transformWith {
+      case Failure(t) =>
+        val msg =
+          s"Unable to retrieve clone workspace results for workspace $workspaceId: unable to retrieve request context for $userEmail"
+        logFailure(msg, Some(t))
+        cloneFail(workspaceId, msg).map(_ => Incomplete)
+      case Success(ctx) =>
+        Try(workspaceManagerDAO.getCloneWorkspaceResult(workspaceId, job.jobControlId.toString, ctx)) match {
+          case Success(result) => handleCloneResult(workspaceId, result)
+          case Failure(t) =>
+            val msg = s"Api call to get clone result from workspace manager failed with: ${t.getMessage}"
+            logFailure(msg, Some(t))
+            cloneFail(workspaceId, msg).map(_ => Incomplete)
+        }
+    }
+
+  }
+
+  def handleCloneResult(workspaceId: UUID, result: CloneWorkspaceResult)(implicit
+    executionContext: ExecutionContext
+  ): Future[JobStatus] = Option(result.getJobReport).map(_.getStatus) match {
+    case Some(JobReport.StatusEnum.RUNNING) => Future.successful(Incomplete)
+    case Some(JobReport.StatusEnum.SUCCEEDED) =>
+      val completeTime = DateTime.parse(result.getJobReport.getCompleted)
+      cloneSuccess(workspaceId, completeTime).map(_ => Complete)
+    // set the error, and indicate this runner is finished with the job
+    case Some(JobReport.StatusEnum.FAILED) =>
+      val msg = Option(result.getErrorReport)
+        .map(_.getMessage)
+        .getOrElse("Cloning failure Reported, but no errors returned")
+      cloneFail(workspaceId, msg).map(_ => Complete)
+    case None =>
+      val msg = Option(result.getErrorReport)
+        .flatMap(report => Option(report.getMessage))
+        .map(errorMessage => s"Cloning Failed: $errorMessage")
+        .getOrElse("No job status or errors returned from workspace cloning")
+      cloneFail(workspaceId, msg).map(_ => Complete)
+  }
+
+  def cloneSuccess(wsId: UUID, finishTime: DateTime)(implicit
+    executionContext: ExecutionContext
+  ): Future[Option[Workspace]] =
+    getWorkspace(wsId).flatMap {
+      case Some(workspace) =>
+        val updatedWS = workspace.copy(completedCloneWorkspaceFileTransfer = Some(finishTime))
+        dataSource.inTransaction(_.workspaceQuery.createOrUpdate(updatedWS)).map(Option(_))
```

Comment:
What does the `.map(Option(_))` do?

Reply:
It just wraps the result of `dataSource.inTransaction(_.workspaceQuery.createOrUpdate(updatedWS))` in an `Option`, so the return type is correct.